src/pages/iou/request/step/IOURequestStepCategory.js

```diff
@@ -50,10 +53,26 @@ function IOURequestStepCategory({
      * @param {String} category.searchText
      */
     const updateCategory = (category) => {
-        if (category.searchText === transaction.category) {
+        const isSelectedCategory = category.searchText === transaction.category;
+        const updatedCategory = isSelectedCategory ? '' : category.searchText;
+
+        // The case edit split bill
+        if (isBillSplit && isEditing) {
+            IOU.setDraftSplitTransaction(transaction.transactionID, {category: category.searchText});
+            navigateBack();
+            return;
+        }
+        // The casse edit request
+        if (isEditing) {
+            IOU.updateMoneyRequestCategory(transaction.transactionID, report.reportID, updatedCategory);
+            Navigation.dismissModal();
+            return;
+        }
+        // The case create request monney or split bill or send money
+        if (isSelectedCategory) {
             IOU.resetMoneyRequestCategory_temporaryForRefactor(transactionID);
         } else {
-            IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, category.searchText);
+            IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, updatedCategory);
         }
```

Comment:
@dukenv0307  Please check this comment https://github.com/Expensify/App/issues/34608#issuecomment-1919249308.